test/verifier/embedded-zkp-verifier.test.ts

```diff
@@ -52,8 +53,146 @@ describe("Embedded ZKP Verifier", function () {
     });
 
     const { inputs, pi_a, pi_b, pi_c } = prepareInputs(proofJson);
+
     const tx = await verifier.submitZKPResponse(0, inputs, pi_a, pi_b, pi_c);
     const txRes = await tx.wait();
+    const receipt = await ethers.provider.getTransactionReceipt(txRes.hash);
+
+    // 2 events are emitted
+    expect(receipt?.logs.length).to.equal(2);
+
+    const dataEvent1 = receipt?.logs[0].data;
```

Comment:
you can call the vars directly as dataBeforeProof / dataAfterProof ... as bit more readability IMHO